std/base.m31

```diff
@@ -12,7 +12,7 @@ handle
   | as_eq _ => None
 end
 
-let (|>) x f = f x
+let (|>) x f : Π α β, α → (α → β) → β = f x
```

Comment:
Before this PR the type of `( |> )` was `list ?α → (list ?α → list ?α) →
list ?α`, which is rather useless. Could we keep track of the reason why we
think something can't be generalised? And try to generalise once we see
enough type variables instantiated to make the applications unambiguous?

On 14 June 2016 at 19:49, Andrej Bauer <notifications@github.com> wrote:

> In std/base.m31
> <https://github.com/Andromedans/andromeda/pull/351#discussion_r67020872>:
>
> > @@ -12,7 +12,7 @@ handle
> >    | as_eq _ => None
> >  end
> >
> > -let (|>) x f = f x
> > +let (|>) x f : Π α β, α → (α → β) → β = f x
>
> Oops, I forgot about the overloading of applications.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Andromedans/andromeda/pull/351/files/3bd0b76b5aca6fc1780cb55e3bbf510bac973f9f#r67020872>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAzuGkasf7wrV8aQdbRJDhuJqLDBskYdks5qLumRgaJpZM4I1ai0>
> .
>